src/host_checks/CountriesContacts.cpp

```diff
@@ -0,0 +1,28 @@
+#include "ntop_includes.h"
+#include "host_checks_includes.h"
+
+CountriesContacts::CountriesContacts() : HostCheck(ntopng_edition_community, false /* All interfaces */, true /* Exclude for nEdge */, false /* NOT only for nEdge */) {
+    countries_contacts_threshold = (u_int32_t)100;
+}
+
+void CountriesContacts::periodicUpdate(Host *h, HostAlert *engaged_alert) {
+    HostAlert *alert = engaged_alert;
+    u_int32_t contacted_countries = 0;
+
+    if ((contacted_countries = getContactedCountries(h)) >= countries_contacts_threshold) {
+        if (!alert) alert = allocAlert(this, h, CLIENT_FAIR_RISK_PERCENTAGE, contacted_countries, countries_contacts_threshold);
+        if (alert) h->triggerAlert(alert);
+    }
+    h->resetCountriesContacts();
```

Comment:
This resets every minute - not 5 minutes. Is this the desired behavior?

Reply:
It was not the desired behavior, in fact we removed the reset from the `periodicUpdate()` and we inserted it in: https: //github.com/fram112/ntopng/blob/f1e590a7678679c86a9934ffc58a62a6d5e7a5b1/src/LocalHostStats.cpp#L703
The counter now resets every 5 minutes.